api/agent.go

```diff
@@ -118,11 +118,12 @@ type AgentServiceConnectProxyConfig struct {
 	DestinationServiceID   string                 `json:",omitempty"`
 	LocalServiceAddress    string                 `json:",omitempty"`
 	LocalServicePort       int                    `json:",omitempty"`
+	Mode                   ProxyMode              `json:",omitempty"`
+	TransparentProxy       TransparentProxyConfig `json:",omitempty"`
```

Comment:
Reminder: `omitempty` doesn't apply to empty structs. This means that because this is not a pointer type every single API result is going to say `..., "TransparentProxy": {}, ...`

Reply:
Yep, was just matching the other fields. Same applies to MeshGateway and Expose below.

Left a comment about this in the main PR description, going to leave this as not-a-pointer for now and going to revisit that for GA. Want to explore some alternatives like using custom JSON marshaling function.